src/main/scala/ing/wbaa/druid/client/DruidAdvancedHttpClient.scala

```diff
@@ -415,20 +418,27 @@ object DruidAdvancedHttpClient extends DruidClientBuilder {
             )
           }
         }
-        .map {
+        .mapAsyncUnordered(parallelism) {
           // consider any response with HTTP Code different from StatusCodes.OK as a failure
           case (triedResponse, responsePromise) =>
             triedResponse match {
               case Success(response) if response.status != StatusCodes.OK =>
-                val failure =
-                  Failure(
-                    new HttpStatusException(
-                      response.status,
-                      response.entity.toStrict(responseParsingTimeout).value.flatMap(_.toOption)
+                response.entity
+                  .toStrict(responseParsingTimeout)
+                  .map(Success(_)) // Success[HttpEntity.Strict] --> Success[Try[HttpEntity.Strict]]
```

Comment:
I find the comments a bit confusing. Maybe it's best just to only specify the parameters:
   .map[Success[HttpEntity.Strict]](...)
   .recover[Try[HttpEntity.Strict]] { ...}
maybe also applies to DruidClient#handleResponse

Reply:
That to me doesn't help with readability. It becomes bracket soup. 😞 

I'll play around with the comments a bit and see if I can come up with something better.

Reply:
Fixed in 13f76c2d72d450e4703a1974526698a7046a6d39